tools/azsdk-cli/docs/specs/0-scenario-2.spec.md

```diff
@@ -0,0 +1,818 @@
+- `--test-filter <pattern>`: Run only tests matching pattern
+- `--parallel`: Run tests in parallel (when supported)
+- `--verbose`: Show detailed test output
+
+**Expected Output:**
+
+```text
+Running tests for Health Deidentification SDK...
+
+✓ Python tests passed (12 recordings captured)
+✓ .NET tests passed (10 recordings captured)
+
+Test execution complete. Recordings saved to language-specific directories.
+```
+
+### 9. Teardown Test Resources
+
+**Command:**
+
+```bash
+azsdk test-resources manage --service healthdataaiservices --action delete
+```
+
+**Options:**
+
+- `--service <name>`: Service short name (required)
+- `--action <create|delete>`: Action to perform (required)
+- `--resource-group <name>`: Override the default resource group naming convention
+- `--subscription <id>`: Target a specific Azure subscription
+
+**Expected Output:**
+
+```text
+Tearing down test resources for healthdataaiservices...
+
+✓ Resource group azsdk-healthdataaiservices-tests deleted
+
+Test resources cleaned up successfully.
+```
+
+---
+
+## Pipeline & CI Usage
+
+Scenario 2 also needs to consider **how the CLI tools will be used inside CI/CD pipelines** (for example, GitHub Actions, Azure Pipelines), even though the exact requirements are still evolving.
+
+### Goals
+
+- Enable **non-interactive**, repeatable runs of the Scenario 2 workflow using CLI commands.
+- Make it easy to plug `azsdk` commands into existing SDK build/test pipelines with minimal glue.
+- Ensure environment setup and customization flows behave predictably in headless environments.
+
+### Early Requirements & Considerations
+
+- `azsdk_package_setup_env` and `azsdk_verify_setup`:
+  - Must support **exit codes** that cleanly distinguish "all good", "fixable issues", and "blocking failures".
+  - Should be configurable via flags or environment variables in CI (for example, `--no-install` or `AZSDK_SETUP_ENV_MODE=verify-only`) to avoid surprise installations on shared agents.
+- Customization and generation:
+  - `azsdk_package_customize_typespec`, `azsdk_package_generate_code`, and `azsdk_package_customized_code_update` should be chainable in a single job with clear, machine-readable logs.
+  - Commands should accept **explicit paths** for specs, workspaces, and output folders to avoid assumptions about checkout layout.
+- Testing:
+  - `azsdk_test_live` is expected to be **opt-in** for CI, guarded behind environment variables or flags and properly parameterized with subscriptions/resource groups.
+  - `azsdk_test_playback` is expected to be **safe to run by default** in CI for regression validation.
+
+### Open Design Questions for Pipelines
+
+These are intentionally left for follow-up design (Wes to clarify):
+
+1. Should there be a **single high-level pipeline command** (for example, `azsdk scenario run --id scenario-2`) that orchestrates all stages, or should pipelines explicitly compose lower-level commands?
```

Comment:
why would we want to specify by scenarios? or are we using scenarios here as an example for like "running on pr" "running on new package" etc.